File: tests/fixtures/process/kotlin/bridge.kt
import java.io.BufferedReader
import java.io.InputStreamReader
import java.io.PrintWriter
import kotlinx.serialization.json.Json
import kotlinx.serialization.json.JsonObject
import kotlinx.serialization.json.JsonPrimitive
import kotlinx.serialization.json.JsonArray
import kotlinx.serialization.json.JsonElement

fun main() {
    val reader = BufferedReader(InputStreamReader(System.`in`))
    val input = reader.readLine() ?: "{}"
    
    try {
        val json = Json.parseToJsonElement(input).jsonObject
        val function = json["function"]?.jsonPrimitive?.content ?: ""
        val args = json["args"]?.jsonArray ?: JsonArray(emptyList())
        
        val result = when (function) {
            "add" -> {
                if (args.size < 2) {
                    mapOf("error" to "add requires 2 arguments")
                } else {
                    val a = toInt(args[0])
                    val b = toInt(args[1])
                    mapOf("result" to (a + b))
                }
            }
            "subtract" -> {
                if (args.size < 2) {
                    mapOf("error" to "subtract requires 2 arguments")
                } else {
                    val a = toInt(args[0])
                    val b = toInt(args[1])
                    mapOf("result" to (a - b))
                }
            }
            "multiply" -> {
                if (args.size < 2) {
                    mapOf("error" to "multiply requires 2 arguments")
                } else {
                    val a = toInt(args[0])
                    val b = toInt(args[1])
                    mapOf("result" to (a * b))
                }
            }
            "greet" -> {
                if (args.isEmpty()) {
                    mapOf("error" to "greet requires 1 argument")
                } else {
                    val name = args[0].jsonPrimitive.contentOrNull ?: args[0].toString()
                    mapOf("result" to "Hello from Kotlin, $name!")
                }
            }
            else -> mapOf("error" to "Unknown function: $function")
        }
        
        val out = PrintWriter(System.out, true)
        out.println(Json.encodeToString(JsonElement.serializer(), JsonObject(result.map { 
            it.key to JsonPrimitive(it.value.toString()) 
        })))
    } catch (e: Exception) {
        val out = PrintWriter(System.out, true)
        out.println("""{"error": "${e.message}"}""")
    }
}

fun toInt(json: JsonElement): Long {
    return when (json) {
        is JsonPrimitive -> json.content.toLongOrNull() ?: json.content.toDoubleOrNull()?.toLong() ?: 0L
        else -> 0L
    }
}

File: tests/fixtures/process/go/bridge.go
package main

import (
	"encoding/json"
	"fmt"
	"os"
)

type Request struct {
	Module   string          `json:"module"`
	Function string          `json:"function"`
	Args     json.RawMessage `json:"args"`
}

type Response struct {
	Result interface{} `json:"result,omitempty"`
	Error  string      `json:"error,omitempty"`
}

func add(args []interface{}) interface{} {
	if len(args) < 2 {
		return map[string]interface{}{"error": "add requires 2 arguments"}
	}
	a := toFloat(args[0])
	b := toFloat(args[1])
	if a == float64(int64(a)) && b == float64(int64(b)) {
		return int64(a) + int64(b)
	}
	return a + b
}

func subtract(args []interface{}) interface{} {
	if len(args) < 2 {
		return map[string]interface{}{"error": "subtract requires 2 arguments"}
	}
	a := toFloat(args[0])
	b := toFloat(args[1])
	if a == float64(int64(a)) && b == float64(int64(b)) {
		return int64(a) - int64(b)
	}
	return a - b
}

func multiply(args []interface{}) interface{} {
	if len(args) < 2 {
		return map[string]interface{}{"error": "multiply requires 2 arguments"}
	}
	a := toFloat(args[0])
	b := toFloat(args[1])
	if a == float64(int64(a)) && b == float64(int64(b)) {
		return int64(a) * int64(b)
	}
	return a * b
}

func greet(args []interface{}) interface{} {
	if len(args) < 1 {
		return map[string]interface{}{"error": "greet requires 1 argument"}
	}
	name := fmt.Sprintf("%v", args[0])
	return fmt.Sprintf("Hello from Go, %s!", name)
}

func toFloat(v interface{}) float64 {
	switch val := v.(type) {
	case float64:
		return val
	case int64:
		return float64(val)
	case json.Number:
		f, _ := val.Float64()
		return f
	default:
		return 0
	}
}

func main() {
	var req Request
	decoder := json.NewDecoder(os.Stdin)
	if err := decoder.Decode(&req); err != nil {
		resp := Response{Error: fmt.Sprintf("Failed to parse request: %v", err)}
		json.NewEncoder(os.Stdout).Encode(resp)
		return
	}

	var args []interface{}
	if err := json.Unmarshal(req.Args, &args); err != nil {
		resp := Response{Error: fmt.Sprintf("Failed to parse args: %v", err)}
		json.NewEncoder(os.Stdout).Encode(resp)
		return
	}

	var result interface{}
	switch req.Function {
	case "add":
		result = add(args)
	case "subtract":
		result = subtract(args)
	case "multiply":
		result = multiply(args)
	case "greet":
		result = greet(args)
	default:
		resp := Response{Error: fmt.Sprintf("Unknown function: %s", req.Function)}
		json.NewEncoder(os.Stdout).Encode(resp)
		return
	}

	if errResult, ok := result.(map[string]interface{}); ok {
		if errMsg, ok := errResult["error"]; ok {
			resp := Response{Error: fmt.Sprintf("%v", errMsg)}
			json.NewEncoder(os.Stdout).Encode(resp)
			return
		}
	}

	resp := Response{Result: result}
	json.NewEncoder(os.Stdout).Encode(resp)
}